### DISCUSSION TITLE
Change argument when injection service

### DISCUSSION THREAD

**Discussion by huuthinh47** [symfony/symfony#45759] [Q&A] [answered]
i am upgrading symfony 2.7 to 5.3. 
At 2.7, i have 3 services:

```php
api_service.class: App\WebBundle\Helper\MyAPI

a_helper:
    class: '%api_service.class%'
    arguments:
        - '@http_client'
        - '%a.apiUri%'
        - '%error.emailTo%'

b_helper:
    class: '%api_service.class%'
    arguments:
        - '@http_client'
        - '%b.apiUri%'
        - '%error.emailTo%'

c_helper:
    class: '%api_service.class%'
    arguments:
        - '@http_client'
        - '%c.apiUri%'
        - '%error.emailTo%'

App\WebBundle\Helper\MyAPI: '@a_helper'

// i call service b_helper like this
$this->container->get('b_helper');
```

Now, at symfony 5.3, i don't know how to call service 'b_helper' by injection "MyAPI". It always call 'a_helper' default.
i saw this link but i don't know how to implement it.
[https://symfony.com/doc/current/service_container.html#explicitly-configuring-services-and-arguments](url)

**Comment by ahundiak** [comment, marked as answer]
Please clarify what `by injection` means.  Are you saying that `$this->container->get('b_helper');` no longer works or are you trying to directly inject b_helper into a constructor:
```
class SomeClass {
    public function __constructor(MyAPI $bHelper)
```
If it is the latter then you probably want to tweak your service definitions to use [named aliases](https://symfony.com/doc/current/service_container/autowiring.html#dealing-with-multiple-implementations-of-the-same-type):
```
App\WebBundle\Helper\MyAPI $bHelper: `@b_helper`
```
Of course ultimately you will probably want to completely revise your service definitions to use up to date autowire techniques.  But these sorts of hacks should get your started.  And `bin/console debug:container` should quickly become a friend.

**Comment by huuthinh47** [comment]
hi @ahundiak 

> Please clarify what by injection means

Yes, i mean the injection as you said.
```php
class SomeClass {
    public function __constructor(MyAPI $bHelper)
```

```php
App\WebBundle\Helper\MyAPI $bHelper: `@b_helper`
```
i don't know about this. This is really helpful for me.

You saved my life. Thank you very very much 💯
